dask/bag/core.py

```diff
@@ -156,12 +158,15 @@ def to_textfiles(b, path, name_function=str, compression='infer',
 
     """
     if isinstance(path, (str, unicode)):
-        if '*' in path:
-            paths = [path.replace('*', name_function(i))
-                     for i in range(b.npartitions)]
-        else:
-            paths = [os.path.join(path, '%s.part' % name_function(i))
-                     for i in range(b.npartitions)]
+        if not '*' in path:
+            path = os.path.join(path, '*.part')
+
+        formatted_names = [name_function(i) for i in range(b.npartitions)]
+        if formatted_names != sorted(formatted_names):
+            raise ValueError("name_function must preserve the order of its input")
```

Comment:
I suggest that we warn instead of raise.  Generally I trust the user to make this decision on their own.